pkg/agent/plugin/workloadattestor/windows/windows_windows_test.go

```diff
@@ -0,0 +1,276 @@
+				},
+			},
+			selectorValues: []string{"user:domain1\\user1", "user_sid:" + sidUser.String(), "enabled_group_sid:" + sidGroup1.String(), "enabled_group:domain1\\group1"},
+			expectCode:     codes.OK,
+		},
+		{
+			name: "OpenProcess error",
+			pq: &fakeProcessQuery{
+				openProcessErr: errors.New("open process error"),
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to open process: open process error",
+		},
+		{
+			name: "OpenProcessToken error",
+			pq: &fakeProcessQuery{
+				openProcessTokenErr: errors.New("open process token error"),
+				handle:              windows.InvalidHandle,
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to open the access token associated with the process: open process token error",
+		},
+		{
+			name: "GetTokenUser error",
+			pq: &fakeProcessQuery{
+				getTokenUserErr: errors.New("get token user error"),
+				handle:          windows.InvalidHandle,
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to retrieve user account information from access token: get token user error",
+		},
+		{
+			name: "GetTokenGroups error",
+			pq: &fakeProcessQuery{
+				getTokenGroupsErr: errors.New("get token groups error"),
+				handle:            windows.InvalidHandle,
+				tokenUser:         &windows.Tokenuser{User: windows.SIDAndAttributes{Sid: sidUser}},
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to retrieve group accounts information from access token: get token groups error",
+		},
+		{
+			name: "LookupAccount failure",
+			pq: &fakeProcessQuery{
+				lookupAccountErr: errors.New("lookup error"),
+				handle:           windows.InvalidHandle,
+				tokenUser:        &windows.Tokenuser{User: windows.SIDAndAttributes{Sid: sidUser}},
+				tokenGroups:      &windows.Tokengroups{Groups: [1]windows.SIDAndAttributes{{Sid: sidGroup1}}},
+				sidAndAttributes: []windows.SIDAndAttributes{{Sid: sidGroup1, Attributes: windows.SE_GROUP_ENABLED}},
+			},
+			selectorValues: []string{"user_sid:" + sidUser.String(), "enabled_group_sid:" + sidGroup1.String()},
+			expectCode:     codes.OK,
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.WarnLevel,
+					Message: "failed to lookup account from user SID",
+					Data: logrus.Fields{
+						"sid":           sidUser.String(),
+						logrus.ErrorKey: "lookup error",
+					},
+				},
+				{
+					Level:   logrus.WarnLevel,
+					Message: "failed to lookup account from group SID",
+					Data: logrus.Fields{
+						"sid":           sidGroup1.String(),
+						logrus.ErrorKey: "lookup error",
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			defer test.logHook.Reset()
```

Comment:
it is easier just creating a new `test` every time a test case is created (and allow us to run in parallel) 
or just call reset before each test case